### PR TITLE
fix: bound direct core text inputs

### LINKED ISSUE #35: [Bug] Bound direct core text inputs before invoking PII analyzers
## Summary

The public `core` text APIs do not enforce an input-size bound before invoking configured analyzers.

At `d486d878bb0178ecb2ff010ef5bc631fdfd01923`:

- `PrivacyService.analyze(...)` / `analyzeDetailed(...)` delegate directly to `PiiAnalysisCoordinator`.
- tokenization, redaction, and `containsPii(...)` paths that perform automatic analysis also reach the same coordinator.
- `PiiAnalysisCoordinator.analyzeDetailed(...)` checks blank input and analyzer availability, but does not reject large source text before calling each `PiiAnalyzer`.
- `PiiAnalyzer.MAX_RESULT_SPANS` bounds returned span count, not source-text size or analyzer work.

By contrast, the Spring AI JSON/text boundary has `PrivacyJsonPayloadTransformer.MAX_PAYLOAD_CHARACTERS = 1_000_000` and rejects oversized payloads before analysis.

This leaves the direct Spring-independent `core` API with materially different resource-safety behavior.

## Why this matters

All built-in analyzers can perform work or allocation proportional to the complete source text before the core regains control:

- `RegexPiiAnalyzer` runs configured matchers over the full input.
- `OpenNlpPiiAnalyzer` tokenizes the full input and allocates token-position / token arrays.
- `PresidioAnalyzer` serializes the full text into an HTTP request body before sending it.
- custom analyzers are explicitly allowed and may allocate based on input size before returning.

The current contract documents that custom analyzers must bound their own work, but that cannot protect built-in analyzers or provide a uniform guardrail when an application exposes direct `PrivacyService` operations to untrusted input.

The threat model also states that the library enforces size/complexity bounds when inspecting supported text, JSON, core value structures, and responses. Direct core text currently does not have an equivalent hard input bound.

## Minimal proof of the missing boundary

A deterministic test can use a counting analyzer:

```java
AtomicInteger calls = new AtomicInteger();
AtomicInteger seenLength = new AtomicInteger();

PiiAnalyzer analyzer = (text, options) -> {
    calls.incrementAndGet();
    seenLength.set(text.length());
    return List.of();
};

PrivacyService service = new PrivacyService(
    List.of(analyzer),
    PiiAnalysisOptions.defaults()
);

service.analyze("x".repeat(10_000_000));

assertThat(calls).hasValue(1);
assertThat(seenLength).hasValue(10_000_000);
```

There is currently no core `PAYLOAD_LIMIT_EXCEEDED` failure before analyzer invocation.

## Expected behavior

Direct text operations should have a documented hard source-text limit enforced by the core before any analyzer receives the text.

The limit should apply consistently to automatic-analysis entry points, including:

- `analyze` / `analyzeDetailed`;
- `tokenize` / `analyzeAndTokenize` when analysis is performed;
- `redact` when analysis is performed;
- `containsPii` when analysis is performed;
- caller-supplied-span operations should also reject an oversized source text before range resolution/transformation.

## Proposed fix

1. Introduce a published core limit such as `MAX_TEXT_INPUT_CHARACTERS` (preferably aligned with the existing Spring AI payload limit unless there is a documented reason not to).
2. Enforce it centrally in `PiiAnalysisCoordinator` / the core source-text validation path so every analyzer is protected before invocation.
3. Return a privacy-safe `PrivacyGuardrailException` with `PAYLOAD_LIMIT_EXCEEDED` and the phase appropriate to the operation.
4. Keep the existing `MAX_TRANSFORMED_TEXT_CHARACTERS` as a separate output-expansion limit; input and output limits serve different purposes.
5. If applications need a different ceiling, expose a bounded configuration object rather than leaving the direct API unlimited.

## Regression tests

Please add tests proving:

- an oversized source text is rejected before a custom analyzer is called;
- Regex, OpenNLP, and Presidio adapters are never invoked for oversized direct-core input;
- caller-supplied-span operations reject oversized source text consistently;
- boundary-size input succeeds and boundary+1 fails with a safe typed error;
- Spring AI and direct-core limits/documentation stay aligned.

## Severity

Suggested priority: **P2 security hardening / contract consistency**. The Spring AI path already has a hard payload limit, but direct public core use can currently hand arbitrarily large untrusted text to analyzers.

### PR DESCRIPTION
**Pull request by NewChoBo**
## Summary

Apply the existing Spring AI payload-size limit to direct core text operations.

Oversized text is rejected before analyzers run or caller-supplied spans are resolved. The Spring AI boundary keeps its current behavior, with both modules using the same limit.

Add regression tests for the boundary and validation order.

## Related Issue

Closes #35

## Verification

- `./gradlew clean check` with Java 17

## Checklist

- [x] I added a `Signed-off-by` line to each commit (`git commit -s`) per the [DCO](https://developercertificate.org/).
- [x] I added focused tests when behavior changed and ran the relevant checks locally.
- [x] I updated any affected documentation and configuration examples.

### CODE REVIEW

File: spring-ai-privacy-guardrails-core/src/main/java/io/github/ultramancode/springai/privacy/core/PiiAnalysisCoordinator.java [hunk 1]
```diff
@@ -324,6 +326,16 @@ private static void requireSuppliedSpanCount(List<PiiSpan> spans) {
         }
     }
 
+    private static void requireTextInputWithinLimit(String text) {
```
Comment (ultramancode) [inline spring-ai-privacy-guardrails-core/src/main/java/io/github/ultramancode/springai/privacy/core/PiiAnalysisCoordinator.java]: Could we check the input limit before the `isBlank()` early returns in `PrivacyTextTransformer.redact(...)` and `containsPii(...)`? Otherwise, oversized whitespace can bypass the hard limit in those paths.

**Comment by ultramancode** [comment]
Thanks for the contribution! Looks good overall. I left one small inline suggestion—could you take a look?

File: spring-ai-privacy-guardrails-core/src/main/java/io/github/ultramancode/springai/privacy/core/PiiAnalysisCoordinator.java (hunk 1 above)
Comment (NewChoBo) [reply spring-ai-privacy-guardrails-core/src/main/java/io/github/ultramancode/springai/privacy/core/PiiAnalysisCoordinator.java]: Good catch. The size check now runs before the blank-input returns in both `redact(...)` paths and `containsPii(...)`. I added a regression test for oversized whitespace too.

Verified with `.\gradlew.bat clean check` on Java 17.

**Comment by ultramancode** [comment]
LGTM. Thanks for your first contribution!
